Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -32,20 +32,19 @@ public class ResourceDictionary : IResourceDictionary, IDictionary<string, objec
 					return;
 
 				_mergedInstance = s_instances.GetValue(_mergedWith, (key) => (ResourceDictionary)Activator.CreateInstance(key));
-				OnValuesChanged(_mergedInstance.ToArray());
+				OnValuesChanged(_mergedInstance);
 			}
 		}
 
 		ICollection<ResourceDictionary> _mergedDictionaries;
 		public ICollection<ResourceDictionary> MergedDictionaries {
-			get {
-				if (_mergedDictionaries == null) {
-					var col = new ObservableCollection<ResourceDictionary>();
-					col.CollectionChanged += MergedDictionaries_CollectionChanged;
-					_mergedDictionaries = col;
-				}
-				return _mergedDictionaries;
-			}
+			get { return _mergedDictionaries ?? (_mergedDictionaries = CreateMergedDictionariesCollection()); }
+		}
+
+		private ICollection<ResourceDictionary> CreateMergedDictionariesCollection() {
```

Comment:
No point in splitting this in a separate method

Reply:
This splitting allows to use more expressive C# 7 code style.
```javascript
		ICollection<ResourceDictionary> _mergedDictionaries;
		public ICollection<ResourceDictionary> MergedDictionaries
		{
			get
			{
				if (_mergedDictionaries == null)
				{
					var col = new ObservableCollection<ResourceDictionary>();
					col.CollectionChanged += MergedDictionaries_CollectionChanged;
					_mergedDictionaries = col;
				}
				return _mergedDictionaries;
			}
			get { return _mergedDictionaries ?? (_mergedDictionaries = CreateMergedDictionariesCollection()); }
		}
```
Current spitted code may converted at next reformatting step into more clean view
```javascript
		ICollection<ResourceDictionary> _mergedDictionaries;
		public ICollection<ResourceDictionary> MergedDictionaries =>
			_mergedDictionaries ?? (_mergedDictionaries = CreateMergedDictionariesCollection());

		private ICollection<ResourceDictionary> CreateMergedDictionariesCollection() {
			var collection = new ObservableCollection<ResourceDictionary>();
			collection.CollectionChanged += MergedDictionaries_CollectionChanged;
			return collection;
		}
```